contrib/pkg/utils/generic.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/openshift/hive/pkg/resource"
+)
+
+type releasePayload struct {
+	PullSpec string `json:"pullSpec"`
+}
+
+func DetermineReleaseImageFromSource(sourceURL string) (string, error) {
+	resp, err := http.Get(sourceURL)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	payload := &releasePayload{}
+	err = json.Unmarshal(data, payload)
+	if err != nil {
+		return "", err
+	}
+	return payload.PullSpec, nil
+}
+
+func GetResourceHelper(command string) (resource.Helper, error) {
```

Comment:
This should take a logger rather than the name of the command.
```suggestion
func GetResourceHelper(logger log.FieldLogger) (resource.Helper, error) {
```

Reply:
You can just create a contextual logger in run(), or I think you could also pass the one from your commands main down into the "Options" struct you have here as a non-exported field, and then use it as o.logger.

Reply:
While explicit logger to be passed down from hiveutil mail would be ideal, I wonder if simply passing an empty context is enough for this case.